### PR TITLE
Fixes stork path param resolution in REST Client

### LINKED ISSUE #37713: PathParam containing "/" character are not well encoded as "%2F" when using reactive rest client with Stork
### Describe the bug

I need to call an external API from my Quarkus backend.
For this example, the external api is like this : http://localhost:8081/api/v1/path/{pathParam}
With in my case "pathParam"=/x/y/z 

When using the reactive rest client, everything is working well and my external API give me a 200.

As soon as i add the stork extension to retrieve the URL of the external API, i got a 404 error from my external API.
Stork is used only to retrieve "http://localhost:8081" as the base url.

If i understand well, when using reactive rest client without stork everything is ok because the called url is :
http://localhost:8081/api/v1/path/x%2Fy%2Fz (with "/" character correctly encoded)

When i do the same with stork, the url is well resolved (http://localhost:8081) but the url called is :
http://localhost:8081/api/v1/path/x/y/z (with "/" character not encoded as "%2F"

I will try to give a reproducer later if needed. Not easy to extract it actually  because it's part of a big project.

### Expected behavior

The expected behaviour would be to have the "/" (or others) well encoded when using stork extension.

### Actual behavior

Actuel behavior is the "/" character as a PathParam is not well encoded only when using stork extension. It work's well when using the reactive rest client without Stork.

### How to Reproduce?

To reproduce :

* Call an external API that contains "/" character in path param. (http://localhost:port/api/v1/path/{pathParam} with pathParam=/x/y/z
* With reactive rest client and without stork -> http response is 200
* With reactive rest client + stork -> http response is 404

### Output of `uname -a` or `ver`

Linux x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.9.1" 

### Quarkus version or git rev

3.6.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.6

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by aureamunoz**
- Fixes #37713

### CODE REVIEW

**Comment by aureamunoz** [comment]
cc @cescoffier

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkDevModeTest.java [hunk 1]
```diff
@@ -30,6 +30,9 @@ public static void setUp() {
         wireMockServer.stubFor(WireMock.get("/hello")
                 .willReturn(aResponse().withFixedDelay(1000)
                         .withBody(WIREMOCK_RESPONSE).withStatus(200)));
+        //        wireMockServer.stubFor(WireMock.get(urlPathTemplate("/hello/{name}"))
```
Comment (cescoffier) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkDevModeTest.java:33]: should it be completely removed?

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java [hunk 2]
```diff
@@ -15,54 +15,65 @@
...
+        assertThat(greeting).isEqualTo("Hello, World!");
 
         greeting = ClientBuilder.newClient().target(UriBuilder.fromUri("stork://hello-service/hello")).request()
                 .get(String.class);
-        assertThat(greeting).isEqualTo("Hello");
+        assertThat(greeting).isEqualTo("Hello, World!");
+
+        greeting = ClientBuilder.newClient().target("stork://hello-service/hello").path("big bird").request()
+                .get(String.class);
+        assertThat(greeting).isEqualTo("Hello, big%20bird");
```
Comment (cescoffier) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java:59]: Why it's not decoded here?

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/StorkClientRequestFilter.java [hunk 3]
```diff
@@ -62,7 +62,8 @@ public void filter(ResteasyReactiveClientRequestContext requestContext) {
                             }
                             // Service instance can also contain an optional path.
                             Optional<String> path = instance.getPath();
-                            String actualPath = uri.getPath();
+                            String actualPath = uri.getRawPath();
+                            //                            extractParameter
```
Comment (cescoffier) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/StorkClientRequestFilter.java:66]: Should it be removed?

**Review by cescoffier — CHANGES_REQUESTED** [review]
it's not clear why the path is still ended when received by resteasy.

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java (hunk 2 above)
Comment (aureamunoz) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java:59]: Resteasy encodes the request's pathParam containing a blank into big%20bird. Stork appends this in the new URI once it knows the actual endpoint. The remote resource receives that (encoded) String `big%20bird`and returns it in the response. The SyncInvoker#get method doesn't decode it so we have the String as it is.

Comment (cescoffier) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java:59]: @geoand is it the expected behavior? I'm surprised that RestEasy Reactive does not decode the path parameter.

Comment (geoand) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java:59]: I don't think it's correct. The parameter should be decoded

Comment (aureamunoz) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java:59]: The current issue is that the Stork filter is currently sending a raw path, thus an encoded big%20bird. The resource receives it so in the path param. How would the resource know that it is a string to be decoded? The encoding/decoding should already have happened. The resource, perceiving the string as it is without the need for decoding, sends it back.

Comment (geoand) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java:59]: Okay, in that case it makes sense

**Comment by aureamunoz** [comment]
I managed to figure out where the encoding were being done. Now it should be ok!

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/StorkClientRequestFilter.java [hunk 4]
```diff
@@ -79,11 +80,12 @@ public void filter(ResteasyReactiveClientRequestContext requestContext) {
                                     }
                                 }
                             }
-
+                            //To avoid the path double encoding we create uri with path=null and set the path after
```
Comment (cescoffier) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/StorkClientRequestFilter.java:83]: ah ah, nice trick!

@geoand FYI.

**Review by cescoffier — APPROVED** [review]
nice catch!

**Comment by cescoffier** [comment]
@aureamunoz can you squash the commits?

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/StorkClientRequestFilter.java (hunk 4 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/StorkClientRequestFilter.java:83]: Very interesting :)

**Review by geoand — APPROVED** [review]

**Pull request merged**